src/fdmdv2_main.cpp

```diff
@@ -2602,18 +2616,24 @@ void MainFrame::OnTogBtnOnOff(wxCommandEvent& event)
 
             assert(g_pfreedv != NULL);
         
-            // init Codec 2 LPC Post Filter (FreeDV 1600)
-
-            codec2_set_lpc_post_filter(freedv_get_codec2(g_pfreedv),
-                                       wxGetApp().m_codec2LPCPostFilterEnable,
-                                       wxGetApp().m_codec2LPCPostFilterBassBoost,
-                                       wxGetApp().m_codec2LPCPostFilterBeta,
-                                       wxGetApp().m_codec2LPCPostFilterGamma);
-
+            // Set buffer size.
+            g_bufferSize = (int)(FRAME_DURATION * freedv_get_speech_sample_rate(g_pfreedv));
+
+            // init Codec 2 LPC Post Filter (FreedV 1600)
+
+            struct CODEC2 *c2 = freedv_get_codec2(g_pfreedv);
+            if (c2 != NULL) {
+                codec2_set_lpc_post_filter(c2,
+                                           wxGetApp().m_codec2LPCPostFilterEnable,
+                                           wxGetApp().m_codec2LPCPostFilterBassBoost,
+                                           wxGetApp().m_codec2LPCPostFilterBeta,
+                                           wxGetApp().m_codec2LPCPostFilterGamma);
+            }
+            
             // Init Speex pre-processor states
```

Comment:
@ac0zj How about this Brad?  I thought this was perhaps easier than adding (yet another) API function.

Reply:
I think that's fine - simple and makes sense!